esp/smc/SMCLib/TpCommon.cpp

```diff
@@ -230,6 +230,37 @@ extern TPWRAPPER_API SecAccessFlags getDZFileScopePermissions(IEspContext& conte
     return getDZPathScopePermissions(context, dropZoneName, dropZonePath, dropZoneHost);
 }
 
+extern TPWRAPPER_API SecAccessFlags getLegacyDZPhysicalPerms(IEspContext& context, const char* dropZoneName, const char* dropZoneHost,
+    const char* dropZoneFile, bool pathOnly)
+{
+    if (isEmptyString(dropZoneHost) && isEmptyString(dropZoneName))
+        throw makeStringException(ECLWATCH_INVALID_INPUT, "Neither dropzone plane or host specified.");
+
+    Owned<IUserDescriptor> userDesc = createUserDescriptor();
+    userDesc->set(context.queryUserId(), context.queryPassword(), context.querySignature());
+
+    StringBuffer host;
+    if (isEmptyString(dropZoneHost))
+    {
+        Owned<IPropertyTree> plane = getDropZonePlane(dropZoneName);
+        if (getPlaneHost(host, plane, 0))
+            dropZoneHost = host.str();
+        else
+            dropZoneHost = "localhost";
+    }
+
+    //The setExternal() requests the path with a file name. The file name is not used by the getDLFNPermissions().
+    //If the dropZoneFile is a path only, the dropZoneScope + a faked file name is passed to the setExternal().
+    StringBuffer pathWithFileName(dropZoneFile);
```

Comment:
it would help if there were 2 functions one for Path and one for File (getLegacyDZPhysicalFilePerms, getLegacyDZPhysicalPathPerms)
And, rather than this workaround for the lack of support from IDistributedFileDirectory for validating a path that isn't a file, it would be better to add support.
What's missing I think is a simple:
```
IDistributedFileDirectory::getScopePermissions(const char *scopes, ....) 
```

method.

If that is added, then the common code for that getLegacyDZPhysicalFilePerms and getLegacyDZPhysicalPathPerms calls , can use a CDfsLogicalFileName and conditionally do:
```
if (isPath)
    scopesToCheck = dlfn.get();
else
{
    dlfn.getScopes(scopes);
    scopesToCheck = scopes;
}
return queryDistributedFileDirectory().getScopePermissions(scopesToCheck, userDesc);

```